modules/openthread/platform/openthread.c

```diff
@@ -0,0 +1,486 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *   This file implements the OpenThread module initialization and state change handling.
+ *
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(net_openthread_platform, CONFIG_OPENTHREAD_PLATFORM_LOG_LEVEL);
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/version.h>
+#include <zephyr/sys/check.h>
+#include <zephyr/net/openthread.h>
+
+#include "platform-zephyr.h"
+
+#include <openthread/child_supervision.h>
+#include <openthread/cli.h>
+#include <openthread/ip6.h>
+#include <openthread/link.h>
+#include <openthread/link_raw.h>
+#include <openthread/ncp.h>
+#include <openthread/message.h>
+#include <openthread/platform/diag.h>
+#include <openthread/tasklet.h>
+#include <openthread/thread.h>
+#include <openthread/dataset.h>
+#include <openthread/joiner.h>
+#include <openthread-system.h>
+#include <utils/uart.h>
+
+#if defined(CONFIG_OPENTHREAD_NAT64_TRANSLATOR)
+#include <openthread/nat64.h>
+#endif /* CONFIG_OPENTHREAD_NAT64_TRANSLATOR */
+
+#define OT_STACK_SIZE (CONFIG_OPENTHREAD_THREAD_STACK_SIZE)
+
+#if defined(CONFIG_OPENTHREAD_THREAD_PREEMPTIVE)
+#define OT_PRIORITY K_PRIO_PREEMPT(CONFIG_OPENTHREAD_THREAD_PRIORITY)
+#else
+#define OT_PRIORITY K_PRIO_COOP(CONFIG_OPENTHREAD_THREAD_PRIORITY)
+#endif
+
+#if defined(CONFIG_OPENTHREAD_NETWORK_NAME)
+#define OT_NETWORK_NAME CONFIG_OPENTHREAD_NETWORK_NAME
+#else
+#define OT_NETWORK_NAME ""
+#endif
+
+#if defined(CONFIG_OPENTHREAD_CHANNEL)
+#define OT_CHANNEL CONFIG_OPENTHREAD_CHANNEL
+#else
+#define OT_CHANNEL 0
+#endif
+
+#if defined(CONFIG_OPENTHREAD_PANID)
+#define OT_PANID CONFIG_OPENTHREAD_PANID
+#else
+#define OT_PANID 0
+#endif
+
+#if defined(CONFIG_OPENTHREAD_XPANID)
+#define OT_XPANID CONFIG_OPENTHREAD_XPANID
+#else
+#define OT_XPANID ""
+#endif
+
+#if defined(CONFIG_OPENTHREAD_NETWORKKEY)
+#define OT_NETWORKKEY CONFIG_OPENTHREAD_NETWORKKEY
+#else
+#define OT_NETWORKKEY ""
+#endif
+
+#if defined(CONFIG_OPENTHREAD_JOINER_PSKD)
+#define OT_JOINER_PSKD CONFIG_OPENTHREAD_JOINER_PSKD
+#else
+#define OT_JOINER_PSKD ""
+#endif
+
+#if defined(CONFIG_OPENTHREAD_PLATFORM_INFO)
+#define OT_PLATFORM_INFO CONFIG_OPENTHREAD_PLATFORM_INFO
+#else
+#define OT_PLATFORM_INFO ""
+#endif
+
+#if defined(CONFIG_OPENTHREAD_POLL_PERIOD)
+#define OT_POLL_PERIOD CONFIG_OPENTHREAD_POLL_PERIOD
+#else
+#define OT_POLL_PERIOD 0
+#endif
+
+#define ZEPHYR_PACKAGE_NAME "Zephyr"
+#define PACKAGE_VERSION     KERNEL_VERSION_STRING
+
+/* Global variables to store the OpenThread module context */
+static otInstance *openthread_instance;
+static struct k_mutex openthread_lock;
```

Comment:
You could use `K_MUTEX_DEFINE` to initialize this statically. Same for `k_work` and `sys_slist_t`.